### ISSUE TITLE
FIM Windows Agent - Case: Restrict.

### ISSUE THREAD

**Issue by damarisg** [wazuh/wazuh-qa#1884]
## Description 

FIM tests has a lot of tests to fix because fails inconsistent times.

### Test Results

| Folder Path | Results |Date | By | Status|
|--|--|--|--|--|
test_files/test_restrict| [ResultsRestrict.zip](https://github.com/wazuh/wazuh-qa/files/7251596/ResultsRestrict.zip)  | 2021/09/29 | Daniela | 🔴  |

| Related to | 
|--|
| https://github.com/wazuh/wazuh-qa/issues/1873

### Pipeline parameters

|  Jenkins branch  | QA branch   | 
|- |- |
|  v4.2.1-rc1 |   1873-4.2.1-fim-windows-agent |

## Packages Info

| Type | Format | Architecture | Versión | Revision |  Tag | File name |
|:--:|:--:|:--:|:--:|:--:|:--:|:--:|
| Server | rpm | x86_64 | v4.2.1 | 40214| v4.2.1 | wazuh-manager-4.2.1-0.1873.x86_64.rpm |
| Agent | rpm | x86_64 | v4.2.1 | 40214|v4.2.1| wazuh-agent-4.2.1-0.1873.x86_64.rpm |
| Agent | msi (Windows) | x86_64 | v4.2.1 | 40214| v4.2.1| wazuh-agent-4.2.1-0.1873.msi |
| Agent | pkg (macOS) | x86_64 | v4.2.1 | 40214| v4.2.1| wazuh-agent-4.2.1-0.1873.pkg |
|Agent | p5p (Solaris 11) | x86_64 | v4.2.1 | 40214|v4.2.1| wazuh-agent_v4.2.1-0.1873-sol11-i386.p5p​|
|Agent | p5p (Solaris 10) | x86_64 | v4.2.1 | 40214|v4.2.1| wazuh-agent_v4.2.1-0.1873-sol10-i386.pkg
|Agent | wpk (Linux) | x86_64 | v4.2.1 | 40214| v4.2.1 | wazuh_agent_v4.2.1-0.1873_linux_x86_64.wpk
|Agent | wpk (Windows) | x86_64 | v4.2.1 | 40214|v4.2.1| wazuh_agent_v4.2.1-0.1873_windows.wpk

**Note**: The tag is `v4.2.1` but for now, to create it on jenkins you need to use `v4.2.1-rc1`.

#### Manager
```
syscheck.debug=2
analysisd.debug=2
monitord.rotate_log=0
```

#### Agent
```
agent.debug=2
syscheck.debug=2
monitord.rotate_log=0
```

### Environment 

Provider| Box| OS| CPU| Memory | 
--|--|--|--|--|
Vagrant | "gusztavvargadr/windows-10" | Windows | 2 | 2048 |

### pytest_args
`-v --tier 0 --tier 1 --tier 2 --fim_mode="realtime" --fim_mode="whodata"`

## Development

Fix `test_fim\test_files\test_restrict\` to Windows. 

If the test work successfully after 2 times, you should see if other member could be verify that it woks successfully. 

## Tasks
- [x] Research about the Test Case.
- [x] Fix the Test Case.
- [x] Verify that each execution works successfully. (Full Green - Attach evidence.)
- [x] Verify that cover `scheduled`, `realtime`, and  `whodata`. Research if this case require all args, or only one.

**Comment by MizugorouZ** [comment]
These are the test results for `test_fim\test_files\test_restrict\` in Windows agent:
| Results | Date | By | Status | Note |
|:--:|:--:|:--:|:--:|:--:|
| [ResultsRestrict_1](https://github.com/wazuh/wazuh-qa/files/7259852/ResultsRestrict_1.zip) | 2021/09/29 | @MizugorouZ | :red_circle: | Executed without disabling any component |
| [ResultsRestrict_2](https://github.com/wazuh/wazuh-qa/files/7259907/ResultsRestrict_2.zip) | 2021/09/29 | @MizugorouZ | :green_circle: | Components disabled manually (sca, rootcheck, syscollector) |
| [ResultsRestrict_3](https://github.com/wazuh/wazuh-qa/files/7259908/ResultsRestrict_3.zip) | 2021/09/29 | @MizugorouZ | :green_circle: | Components disabled with `test_fim\test_files\test_restrict\data\wazuh_conf.yaml` |

**Comment by damarisg** [comment]
I attach the result after it merged, to verify that it continues working. 

| Results | Date | By | Status | 
|:--:|:--:|:--:|:--:|
|[ResultsRestrict](https://github.com/wazuh/wazuh-qa/files/7269580/ResultsRestrict.zip) | 2021/10/01| Seyla | :green_circle: |

Closed by #1958
